src/ado/repscan.ado

```diff
@@ -0,0 +1,152 @@
+
+
+cap program drop repscan
+program define repscan, rclass
+
+{
+    
+    version 14.1
+    
+    // take do-file as argument
+    args do_file
+    
+    // reading file
+    file open myfile using "`do_file'", read
+    display("Scanning file `do_file':")
+    file read myfile line
+    
+    //iterating through lines, saving lines in rows
+    local n_line = 1
+    while r(eof) == 0 {
+            
+        display("    Scanning line `n_line'...")
+        
+        // checking possible reproducibility issues
```

Comment:
Consider keeping track of if you are in a comment of not here. That would reduce a lot of false positives